NotSerializableException with ViewScope$DestructionCallbackWrapper after upgrading SpringBoot to 3.4.0

Joinfaces: 5.4.0
SpringBoot: 3.4.0
spring-integration: 6.4.0

After upgrading to SpringBoot 3.4.0 and while using @ViewScope I get an error:

`Caused by: java.io.NotSerializableException: org.joinfaces.viewscope.ViewScope$DestructionCallbackWrapper`

I am using Spring Session to store session data in Redis which requires classes to be serialized.

Before upgrading `ViewScope.registerDestructionCallback()` was not called and a `ViewScope.DestructionCallbackWrapper` was not created. But after upgrading `registerDestructionCallback()` is now called as spring-integration introduced the `ControlBusCommandRegistry` with spring-integration 6.4.0 which implements `DestructionAwareBeanPostProcessor`.

I was able avoid the error by adding `implements Serializable` to the static classes ViewScope.DestructionCallbackWrapper and ViewScope.SessionListener.

Would it be possible to implement Serializable for these static inner classes?

Stacktrace:
```java
2024-12-12 12:55:31,907 [http-nio-8080-exec-1]    ERROR org.apache.catalina.core.ContainerBase.[Tomcat].[localhost] - Exception Processing [ErrorPage[exceptionType=java.lang.Throwable, location=/pages/error/error.xhtml]]
org.springframework.data.redis.serializer.SerializationException: Cannot serialize
	at org.springframework.data.redis.serializer.JdkSerializationRedisSerializer.serialize(JdkSerializationRedisSerializer.java:97)
	at org.springframework.data.redis.core.AbstractOperations.rawHashValue(AbstractOperations.java:206)
	at org.springframework.data.redis.core.DefaultHashOperations.putAll(DefaultHashOperations.java:161)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.springframework.data.redis.core.BoundOperationsProxyFactory$BoundOperationsMethodInterceptor.doInvoke(BoundOperationsProxyFactory.java:177)
	at org.springframework.data.redis.core.BoundOperationsProxyFactory$BoundOperationsMethodInterceptor.invoke(BoundOperationsProxyFactory.java:148)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
	at org.springframework.data.projection.DefaultMethodInvokingMethodInterceptor.invoke(DefaultMethodInvokingMethodInterceptor.java:69)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:223)
	at jdk.proxy2/jdk.proxy2.$Proxy239.putAll(Unknown Source)
	at org.springframework.session.data.redis.RedisIndexedSessionRepository$RedisSession.saveDelta(RedisIndexedSessionRepository.java:903)
	at org.springframework.session.data.redis.RedisIndexedSessionRepository$RedisSession.save(RedisIndexedSessionRepository.java:891)
	at org.springframework.session.data.redis.RedisIndexedSessionRepository.save(RedisIndexedSessionRepository.java:491)
	at org.springframework.session.data.redis.RedisIndexedSessionRepository.save(RedisIndexedSessionRepository.java:267)
	at org.springframework.session.web.http.SessionRepositoryFilter$SessionRepositoryRequestWrapper.commitSession(SessionRepositoryFilter.java:229)
	at org.springframework.session.web.http.SessionRepositoryFilter.doFilterInternal(SessionRepositoryFilter.java:145)
	at org.springframework.session.web.http.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:82)
	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:362)
	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:278)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:101)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:101)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:164)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:140)
	at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:632)
	at org.apache.catalina.core.ApplicationDispatcher.processRequest(ApplicationDispatcher.java:410)
	at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:330)
	at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:267)
	at org.apache.catalina.core.StandardHostValve.custom(StandardHostValve.java:362)
	at org.apache.catalina.core.StandardHostValve.throwable(StandardHostValve.java:288)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:149)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:344)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:397)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:905)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1741)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
	at java.base/java.lang.Thread.run(Thread.java:1575)
Caused by: org.springframework.core.serializer.support.SerializationFailedException: Failed to serialize object using DefaultSerializer
	at org.springframework.core.serializer.support.SerializingConverter.convert(SerializingConverter.java:64)
	at org.springframework.core.serializer.support.SerializingConverter.convert(SerializingConverter.java:33)
	at org.springframework.data.redis.serializer.JdkSerializationRedisSerializer.serialize(JdkSerializationRedisSerializer.java:95)
	... 48 common frames omitted
Caused by: java.io.NotSerializableException: org.joinfaces.viewscope.ViewScope$DestructionCallbackWrapper
	at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1201)
	at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:359)
	at java.base/java.util.HashMap.internalWriteEntries(HashMap.java:1953)
	at java.base/java.util.HashMap.writeObject(HashMap.java:1506)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:1084)
	at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1530)
	at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1452)
	at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1195)
	at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:359)
	at java.base/java.util.LinkedHashMap.internalWriteEntries(LinkedHashMap.java:421)
	at java.base/java.util.HashMap.writeObject(HashMap.java:1506)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:1084)
	at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1530)
	at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1452)
	at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1195)
	at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1586)
	at java.base/java.io.ObjectOutputStream.defaultWriteObject(ObjectOutputStream.java:452)
	at java.base/java.util.Collections$SynchronizedMap.writeObject(Collections.java:3001)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:1084)
	at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1530)
	at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1452)
	at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1195)
	at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:359)
	at org.springframework.core.serializer.DefaultSerializer.serialize(DefaultSerializer.java:46)
	at org.springframework.core.serializer.Serializer.serializeToByteArray(Serializer.java:56)
	at org.springframework.core.serializer.support.SerializingConverter.convert(SerializingConverter.java:60)
	... 50 common frames omitted```

--- a/joinfaces/src/main/java/org/joinfaces/viewscope/ViewScope.java
+++ b/joinfaces/src/main/java/org/joinfaces/viewscope/ViewScope.java
@@ -16,6 +16,7 @@
 
 package org.joinfaces.viewscope;
 
+import java.io.Serializable;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
@@ -150,7 +151,7 @@ public class ViewScope implements Scope {
 	 * @see org.springframework.web.context.request.DestructionCallbackBindingListener
 	 */
 	@Getter
-	static class SessionListener implements HttpSessionBindingListener {
+	static class SessionListener implements HttpSessionBindingListener, Serializable {
 
 		private final List<DestructionCallbackWrapper> destructionCallbackWrappers = new ArrayList<>();
 
@@ -186,7 +187,7 @@ public class ViewScope implements Scope {
 	 * @author Lars Grefer
 	 * @see #registerDestructionCallback(String, Runnable)
 	 */
-	class PreDestroyViewMapListener implements ViewMapListener {
+	class PreDestroyViewMapListener implements ViewMapListener, Serializable {
 
 		@Override
 		public void processEvent(SystemEvent event) {
@@ -211,7 +212,7 @@ public class ViewScope implements Scope {
 	 * @author Lars Grefer
 	 * @see #registerDestructionCallback(String, Runnable)
 	 */
-	static class DestructionCallbackWrapper {
+	static class DestructionCallbackWrapper implements Serializable {
 
 		@Getter
 		private final String beanName;

